src/System.Net.Http/src/System/Net/Http/HttpClientHandler.Net46.cs

```diff
@@ -0,0 +1,1193 @@
+            {
+                webRequest.Proxy.Credentials = _defaultProxyCredentials;
+            }
+
+            // For Extensibility
+            InitializeWebRequest(request, webRequest);
+
+            return webRequest;
+        }
+
+        // Used to map the ServerCertificateCustomValidationCallback which uses Func<T> to the
+        // HttpWebRequest based RemoteCertificateValidationCallback delegate type.
+        private bool ServerCertificateValidationCallback(
+            object sender,
+            X509Certificate certificate,
+            X509Chain chain,
+            SslPolicyErrors sslPolicyErrors)
+        {
+            return _serverCertificateCustomValidationCallback(
+                null, // TODO Issue #11100: How to map sender (which is an HttpWebRequest) to an HttpRequestMessage?
+                (X509Certificate2)certificate, // This cast will usually always be safe.
+                chain,
+                sslPolicyErrors);
+        }
+
+        // Needs to be internal so that WebRequestHandler can access it from a different assembly.
+        internal virtual void InitializeWebRequest(HttpRequestMessage request, HttpWebRequest webRequest)
+        {
+        }
+
+        private void SetDefaultOptions(HttpWebRequest webRequest)
+        {
+            webRequest.Timeout = Timeout.Infinite; // Timeouts are handled by HttpClient.
+
+            webRequest.AllowAutoRedirect = _allowAutoRedirect;
+            webRequest.AutomaticDecompression = _automaticDecompression;
+            webRequest.PreAuthenticate = _preAuthenticate;
+
+            if (_useDefaultCredentials)
+            {
+                webRequest.UseDefaultCredentials = true;
+            }
+            else
+            {
+                webRequest.Credentials = _credentials;
+            }
+
+            if (_allowAutoRedirect)
+            {
+                webRequest.MaximumAutomaticRedirections = _maxAutomaticRedirections;
+            }
+
+            if (_useProxy)
+            {
+                // If 'UseProxy' is true and 'Proxy' is null (default), let HWR figure out the proxy to use. Otherwise
+                // set the custom proxy.
+                if (_proxy != null)
+                {
+                    webRequest.Proxy = _proxy;
+                }
+            }
+            else
+            {
+                // The use explicitly specified to not use a proxy. Set HWR.Proxy to null to make sure HWR doesn't use
+                // a proxy for this request.
+                webRequest.Proxy = null;
+            }
+
+            if (_useCookies)
+            {
+                webRequest.CookieContainer = _cookieContainer;
+            }
+
+            if (_clientCertOptions == ClientCertificateOption.Automatic && ComNetOS.IsWin7orLater)
+            {
+                X509CertificateCollection automaticClientCerts
+                    = UnsafeNclNativeMethods.NativePKI.FindClientCertificates();
```

Comment:
Where is this method coming from?  Does this assembly have InternalsVisibleTo access to this support in System.dll in desktop?